test/e2e/replica_set.go

```diff
@@ -249,3 +267,87 @@ func rsConditionCheck(f *framework.Framework) {
 	}
 	Expect(err).NotTo(HaveOccurred())
 }
+
+func testAdoptMatchingOrphans(f *framework.Framework) {
+	name := "pod-adoption"
+	By(fmt.Sprintf("Given a Pod with a 'name' label %s is created", name))
+	p := f.PodClient().CreateSync(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  name,
+					Image: nginxImageName,
+				},
+			},
+		},
+	})
+
+	By(fmt.Sprintf("When a replicaset with a matching selector is created"))
+	replicas := int32(2)
+	rsSt := newRSWithSelector(name, replicas, map[string]string{"name": name}, name, nginxImageName, name)
+	rs, err := f.ClientSet.Extensions().ReplicaSets(f.Namespace.Name).Create(rsSt)
+	Expect(err).NotTo(HaveOccurred())
+	// Cleanup the ReplicaSet when we are done.
+	defer func() {
+		if err := framework.DeleteReplicaSet(f.ClientSet, f.InternalClientset, f.Namespace.Name, rs.Name); err != nil {
+			framework.Logf("Failed to cleanup ReplicaSet %v: %v.", rs.Name, err)
+		}
+	}()
+
+	By("Then the orphan pod is adopted")
+	// List the replicaset pods, making sure we observe all the replicas.
+	pods, err := framework.PodsCreated(f.ClientSet, f.Namespace.Name, rs.Name, replicas)
+	Expect(err).NotTo(HaveOccurred())
+	for _, pod := range pods.Items {
+		if pod.DeletionTimestamp != nil {
+			continue
+		}
+		if pod.Name == p.Name && pod.UID == p.UID {
+			// Orphan was adopted
+			return
+		}
+	}
+	Fail("Orphan was not adopted")
+}
+
+func testReleaseControlledNotMatching(f *framework.Framework) {
+	name := "pod-release"
+	By("Given a ReplicaSet is created")
+	replicas := int32(2)
+	rsSt := newRSWithSelector(name, replicas, map[string]string{"name": name}, name, nginxImageName, name)
+	rs, err := f.ClientSet.Extensions().ReplicaSets(f.Namespace.Name).Create(rsSt)
+	Expect(err).NotTo(HaveOccurred())
+	// Cleanup the ReplicaSet when we are done.
+	defer func() {
+		if err := framework.DeleteReplicaSet(f.ClientSet, f.InternalClientset, f.Namespace.Name, rs.Name); err != nil {
+			framework.Logf("Failed to cleanup ReplicaSet %v: %v.", rs.Name, err)
+		}
+	}()
+
+	By("When the matched label of one of its pods change")
+	pods, err := framework.PodsCreated(f.ClientSet, f.Namespace.Name, rs.Name, replicas)
+	Expect(err).NotTo(HaveOccurred())
+	p := pods.Items[0]
+	framework.RunKubectlOrDie("patch", "pod", p.Name, "-n", f.Namespace.Name, "-p", "{\"metadata\":{\"labels\":{\"name\":\"not-matching-name\"}}}")
```

Comment:
I think directly using the client call is better. Something like `f.Clientset.Core().Pods(p.Name).Patch(...)`. We want to limit the number of involved components in the test, so let's not use kubectl.